sw/host/opentitanlib/src/dif/lc_ctrl.rs

```diff
@@ -29,6 +30,196 @@ with_unknown! {
         Scrap = bindgen::dif::dif_lc_ctrl_state_kDifLcCtrlStateScrap as u32,
         PostTransition = bindgen::dif::dif_lc_ctrl_state_kDifLcCtrlStatePostTransition as u32,
         Escalate = bindgen::dif::dif_lc_ctrl_state_kDifLcCtrlStateEscalate as u32,
-        StateInvalide = bindgen::dif::dif_lc_ctrl_state_kDifLcCtrlStateInvalid as u32,
+        StateInvalid = bindgen::dif::dif_lc_ctrl_state_kDifLcCtrlStateInvalid as u32,
+    }
+}
+
+impl DifLcCtrlState {
+    /// Encode the given life cycle state in a redundant format where the
+    /// five-bit value is repeated six times.
+    pub fn redundant_encoding(&self) -> u32 {
+        let value: u32 = self.0;
+        assert_eq!(value & 0b11111, value);
+        (0..6).fold(0u32, |acc, _| (acc << 5) | value)
+    }
+}
+
+pub struct DifLcCtrlToken(bindgen::dif::dif_lc_ctrl_token);
+
+impl From<[u8; 16]> for DifLcCtrlToken {
+    fn from(bytes: [u8; 16]) -> Self {
+        DifLcCtrlToken(bindgen::dif::dif_lc_ctrl_token { data: bytes })
+    }
+}
+
+impl DifLcCtrlToken {
+    /// Converts a 128-bit transition token into four native u32 words. These
+    /// values are suitable to write to [LcCtrlReg::TransitionToken0] and
+    /// friends.
+    pub fn into_register_values(self) -> [u32; 4] {
+        let mut out_words = [0u32; 4];
+        let bytes = self.0.data;
+        bytes
+            .chunks_exact(std::mem::size_of::<u32>())
+            .map(|chunk| u32::from_ne_bytes(chunk.try_into().unwrap()))
```

Comment:
On second thought, this should probably be `u32::from_le_bytes` to match the device's endianness.